python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,531 @@
+        can be the name of a file or a item name in Nuke or Houdini, or a
+        combination of both. This value is displayed to the user and should make
+        it easy for the user to identify what the item represents in terms of
+        the current session or a file on disk.
+
+        .. image:: ./resources/create_item_args.png
+
+        |
+
+        :param str item_type: Item type, typically following a hierarchical dot
+            notation.
+        :param str display_type: Equivalent to the type, but for display
+            purposes.
+        :param str name: The name to represent the item in a UI. This can be a
+            item name in a DCC or a file name.
+        """
+
+        child_item = PublishItem(
+            name,
+            item_type,
+            display_type,
+            parent=self
+        )
+        self._children.append(child_item)
+
+        return child_item
+
+    def add_task(self, plugin):
+        """
+        Create a new publish task and attach it to this item.
+
+        :param plugin: The plugin instance this task will represent/execute.
+        """
+
+        # create the task item and add it to the tree
+        child_task = PublishTask(plugin, self)
+        self._tasks.append(child_task)
+
+        return child_task
+
+    def clear_tasks(self):
+        """
+        Clear all tasks for this item.
+        """
+        self._tasks = []
+
+    def get_property(self, name, default_value=None):
+        """
+        This is a convenience method that will retrieve a property set on the
+        item.
+
+        If the property was set via :meth:`PublishItem.local_properties`, then
+        that will be returned. Otherwise, the value set via
+        :meth:`Item.properties` will be returned. If the property is not set on
+        the item, then the supplied ``default_value`` will be returned (default
+        is ``None``).
+
+        :param name: The property to retrieve.
+        :param default_value: The value to return if the property is not set on
+            the item.
+
+        :return: The value of the supplied property.
+        """
+
+        local_properties = self._get_local_properties()
+
+        return local_properties.get(name) or self.properties.get(name) or \
+            default_value
+
+    @property
+    def active(self):
+        """
+        Returns the item's active state if it has been explicitly set, `None``
+        otherwise
+        """
+        return self._active
+
+    @active.setter
+    def active(self, active_state):
+        """
+        Explicitly set the active state.
+
+        There are 3 active states that can be supplied:
+
+        * ``True``: Set the item to be active
+        * ``False``: Set the item to be inactive
+        * ``None``: Clear the item's state, rely on inheritance within the tree
+        """
+        self._active = active_state
+
+    @property
+    def children(self):
+        """Return the children of this item."""
```

Comment:
I don't have a strong opinion. What do you think?

Reply:
Well, if people are generally just going to iterate over them, `yield` has the benefit of never running the risk of having people of removing a children from the array instead of invoking our `remove_child` method.

Reply:
It's not something we've been careful of in the past, but I would love if we could try to be more careful about in the future.

Especially since in Python 3 puts a heavy emphasis on iterators.